File: src/main/java/org/jetbrains/research/groups/ml_methods/code_completion_benchmark/core/counter/storage/ArrayStorage.java
package org.jetbrains.research.groups.ml_methods.code_completion_benchmark.core.counter.storage;

import java.util.List;

class ArrayStorage {
    static boolean checkExactSequence(List<Integer> indices, int index, int[] successor) {
        boolean valid = successor.length == indices.size() - index;
        if (valid) {
            for (int i = 1; i < successor.length; i++) {
                if (indices.get(index + i) != successor[i]) {
                    valid = false;
                    break;
                }
            }
        }
        return valid;
    }

    static boolean checkPartialSequence(List<Integer> indices, int index, int[] successor) {
        boolean valid = successor.length >= indices.size() - index;
        if (valid) {
            for (int i = 1; i < indices.size() - index; i++) {
                if (indices.get(index + i) != successor[i]) {
                    valid = false;
                    break;
                }
            }
        }
        return valid;
    }
}

File: src/main/kotlin/org/jetbrains/research/groups/ml_methods/code_completion_benchmark/core/counter/storage/ArrayStorage.kt
package org.jetbrains.research.groups.ml_methods.code_completion_benchmark.core.counter.storage

object ArrayStorage {

    fun checkExactSequence(indices: List<Int>, index: Int, successor: IntArray): Boolean {
        var valid = successor.size == indices.size - index
        if (valid) {
            for (i in 1 until successor.size) {
                if (indices[index + i] != successor[i]) {
                    valid = false
                    break
                }
            }
        }
        return valid
    }

    fun checkPartialSequence(indices: List<Int>, index: Int, successor: IntArray): Boolean {
        var valid = successor.size >= indices.size - index
        if (valid) {
            for (i in 1 until indices.size - index) {
                if (indices[index + i] != successor[i]) {
                    valid = false
                    break
                }
            }
        }
        return valid
    }
}